libgalois/src/SharedMemSys.cpp

```diff
@@ -44,7 +44,7 @@ katana::SharedMemSys::SharedMemSys() : impl_(std::make_unique<Impl>()) {
   if (auto init_good = tsuba::Init(&comm_backend); !init_good) {
     KATANA_LOG_FATAL("tsuba::Init: {}", init_good.error());
   }
-  katana::ProgressTracer::SetProgressTracer(katana::JSONTracer::Make());
+  katana::ProgressTracer::Set(katana::TextTracer::Make());
```

Comment:
Suggestion: We could replace the  `SharedMemSys()` with `SharedMemSys(std::unique_ptr<ProgressTracer> tracer = katana::TextTracer::Make())` which would make it explicit to new callers that they can initialize this here.

Reply:
That's a good idea. Ditto for `DistMemSys`.

Reply:
That wouldn't work for distributed tracers we need to pass in the number of hosts and host id to the tracer, which we wouldn't have access to before DistMemSys is called
So I could do that here, but not in enterprise

Reply:
How about doing it in the `NetworkInterface`?

Reply:
Sure sounds good, thanks for the feedback

Reply:
Well that's still kind of the same issue actually, since we are trying to pass in a tracer to the initialization code, but the parameters to create the tracer can't be accessed until during the initialization

I think it's safe to initialize a TextTracer and then have a helper function to enable tracing like StartDistributedTrace
And just not have it as a parameter to DistMemSys, but make it a parameter here